src/saltext/vmware/utils/connect.py

```diff
@@ -23,83 +23,90 @@
-        available to pillar and ext_pillar rendering. These pillar variables
+    config
+        (optional) If specified, allows for a dictionary of data to be made
+        available to pillar and ext_pillar rendering. These variables
         will also override any variables of the same name in pillar or
         ext_pillar.
 
     esxi_host
         (optional) If specified, retrieves the configured username and password for this host.
 
-    Pillar Example:
+    profile
+        Profile to use (optional)
 
-    .. code-block::
+    Pillar Example:
 
-        vmware_config:
-            host: 198.51.100.100
-            password: ****
-            user: @example.com
+    .. code-block:: yaml
 
-        vmware_config:
+        saltext.vmware:
             host: 198.51.100.100
-            password: ****
-            user: @example.com
+            password: CorrectHorseBatteryStaple
+            user: admin@example.com
             esxi_host:
                 198.52.100.105:
                     user: admin
-                    password: ***
+                    password: CorrectHorseBatteryStaple
                 198.52.100.106:
                     user: admin
-                    password: ***
+                    password: CorrectHorseBatteryStaple
+
+    If configuration for multiple VMware services instances is required, they can be
+    set up as different configuration profiles:
+
+    .. code-block:: yaml
+
+        saltext.vmware:
+          profile1:
+            host: 198.51.100.100
+            password: CorrectHorseBatteryStaple
+            user: admin@example.com
+          profile2:
+            host: 198.51.100.100
+            password: CorrectHorseBatteryStaple
+            user: admin@example.com
+            esxi_host:
+              198.52.100.105:
+                user: admin
+                password: CorrectHorseBatteryStaple
+              198.52.100.106:
+                user: admin
+                password: CorrectHorseBatteryStaple
 
     """
     ctx = ssl._create_unverified_context()
-    opts = opts or {}
-    pillar = pillar or {}
-    host = (
-        esxi_host
-        or os.environ.get("VMWARE_CONFIG_HOST")
-        or opts.get("vmware_config", {}).get("host")
-        or pillar.get("vmware_config", {}).get("host")
-    )
-    user, password = get_username_password(esxi_host=host, opts=opts, pillar=pillar)
-    config = {
-        "host": host,
-        "password": password,
-        "user": user,
-    }
+    config = config or {}
+
+    config = get_config(config=config, profile=profile, esxi_host=esxi_host)
+
     if config["host"] is None or config["password"] is None or config["user"] is None:
         raise ValueError("Cannot create service instance, VMware credentials incomplete.")
+
     service_instance = connect.SmartConnect(
-        host=config["host"],
-        user=config["user"],
-        pwd=config["password"],
+        host=config.get("host"),
+        user=config.get("username"),
```

Comment:
**suggestion** this should also be changed to `user` - see the return from `def get_config`

Reply:
That's strange that I missed that. I'll fix that in a few minutes

Reply:
Done